src/HTTP/Cookies/Cookie.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace ILIAS\HTTP\Cookies;
+
+/**
+ * Interface Cookie
+ *
+ * ILIAS cookie representation.
+ * All implementations must be immutable.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @package ILIAS\HTTP\Cookies
+ * @since   5.2
+ * @version 1.0.0
+ */
+interface Cookie {
+
+	/**
+	 * Cookie name.
+	 *
+	 * @return string
+	 */
+	public function getName();
+
+
+	/**
+	 * Cookie value.
+	 *
+	 * @return string|null
+	 */
+	public function getValue();
+
+
+	/**
+	 * Expiration date as unix timestamp.
+	 *
+	 * @return int
+	 */
+	public function getExpires();
+
+
+	/**
+	 * Max age measured in seconds.
+	 * If the max age is zero no max age is set.
+	 *
+	 * @return int
+	 */
+	public function getMaxAge();
+
+
+	/**
+	 * Cookie path.
+	 *
+	 * @return string
+	 */
+	public function getPath();
+
+
+	/**
+	 * Cookie domain.
+	 *
+	 * @return string
+	 */
+	public function getDomain();
+
+
+	/**
+	 * True if it's secure cookie otherwise false.
+	 *
+	 * @return bool
+	 */
+	public function getSecure();
+
+
+	/**
+	 * True if the cookie is http only otherwise false.
+	 *
+	 * @return bool
+	 */
+	public function getHttpOnly();
+
+
+	/**
+	 * Sets the cookie value.
+	 *
+	 * @param null|string $value The cookie value.
+	 *
+	 * @return Cookie
+	 */
+	public function withValue($value = null);
+
+
+	/**
+	 * Sets the expiration date of the cookie.
+	 * If the cookie should be expired please use the expire function.
+	 *
+	 * @param null|\DateTimeInterface|int|string $expires The expiration time of the Cookie.
+	 *
+	 * @return Cookie
+	 */
+	public function withExpires($expires = null);
```

Comment:
A cookie can have a date of expiry or a max-age or both which is a bit confusing but possible. The null parameter is used to erase the date of expiry out of the cookie. The expires method is used set the date of expiry to a date in past. I will update the doc comment to clarify the usage of the method.